actionpack/test/controller/integration_test.rb

```diff
@@ -1122,3 +1122,37 @@ def test_cookies_set_in_setup_are_persisted_through_the_session
     assert_equal({"user_name"=>"david"}, cookies.to_hash)
   end
 end
+
+class IntegrationRequestEncodersTest < ActionDispatch::IntegrationTest
+  class FooController < ActionController::Base
+    def foos
+      render plain: 'ok'
+    end
+  end
+
+  def test_encoding_as_json
+    assert_encoded_as :json, content_type: 'application/json'
+  end
+
+  def test_registering_custom_encoder
+    ActionDispatch::IntegrationTest.register_encoder :wibble, format: 'wibble',
+      content_type: 'text/wibble', param_encoder: -> params { params.to_s }
+
+    assert_encoded_as :wibble, content_type: 'text/wibble',
+      parsed_parameters: Hash.new # Unregistered MIME Type can't be parsed
```

Comment:
Am I right in making this claim? Definitely feels like it could use some shine, but don't have a fully formed idea of which direction to take it.